ocaml-lsp-server/test/e2e-new/code_actions.ml

```diff
@@ -599,3 +599,98 @@ let f (x : t) = x
       "kind": "inferred_intf",
       "title": "Insert inferred interface"
     } |}]
+
+let position_of_offset src x =
+  assert (0 <= x && x < String.length src);
+  let cnum = ref 0
+  and lnum = ref 0 in
+  for i = 0 to x - 1 do
+    if src.[i] = '\n' then (
+      incr lnum;
+      cnum := 0)
+    else incr cnum
+  done;
+  Position.create ~character:!cnum ~line:!lnum
+
+let parse_selection src =
+  let start_pos =
+    match String.index src '$' with
+    | Some x -> x
+    | None -> failwith "expected a selection opening mark"
```

Comment:
Is it possible the user can encounter this error?

1. If yes, then it should be a Jsonrpc error.
2. if not, it should it be a code error.

Reply:
This is part of the test harness. It gets raised when the test input doesn't have $ markers for the selection. Do you want me to use a different exception?

Reply:
Oh never mind. It's fine.